src/PostgREST/Config.hs

```diff
@@ -152,10 +158,10 @@ readCLIShowHelp = customExecParser parserPrefs opts
         long "dump-config" <>
         help "Dump loaded configuration and exit"
       ) <*>
-      strArgument (
+      optional (strArgument (
```

Comment:
So now that this is optional, when running the binary without a config file, I get:

```
postgrest-run

Error parsing config file:
        missing key: db-anon-role
```

Before we would get the "missing" and the `help` output:

```
postgrest
Missing: FILENAME

Usage: postgrest FILENAME
  PostgREST 7.0.0 (UNKNOWN) / create a REST API to an existing Postgres database
...
```

Do you think we should keep it like this(it'd be a breaking change)? Or perhaps we can show the `help` output when no env var is defined?

Reply:
I agree that when no filename is given and no required configs are set via environment variables, we should show the usage screen. It wouldn't need to contain "Missing: FILENAME" (because it's not required) but tell the user to either set a config file or use environment variables.

Once any of the required configs is set or a file name is missing, we can still error out with the same error as right now. The user should know by now **how** to set this up, they just forgot one of the settings.

Reply:
>  "Missing: FILENAME" (because it's not required) but tell the user to either set a config file or use environment variables.

True. A friendly message for that would be great. Also forgot that the help message is also valuable for showing the postgrest version. We definitely need to keep it.

Reply:
> I agree that when no filename is given and no required configs are set via environment variables, we should show the usage screen. It wouldn't need to contain "Missing: FILENAME" (because it's not required) but tell the user to either set a config file or use environment variables.

I did exactly that. I removed the example config file from the usage screen as well and added another `--example` cli option to show it instead.